Simple-Projects/Simple_Projects/Car/Car.cpp

```diff
@@ -47,18 +47,19 @@ void Car::drive(double x, double y) {
 }
 
 void Car::refill(double fuel) {
-    double newFuelAmount = fuelAmount + fuel;
-    fuelAmount = newFuelAmount;
+    const double newFuelAmount = fuelAmount + fuel;
```

Comment:
Note that the change to fuel amount will only be done _after_ the check passes. A thrown exception will leave the object in a valid state